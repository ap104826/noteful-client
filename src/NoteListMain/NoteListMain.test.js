import React from 'react';
import { shallow } from 'enzyme'
import toJson from 'enzyme-to-json'
import LinkListMain from './LinkListMain'

describe(`LinkListMain component`, () => {
  it('renders a .LinkListMain by default', () => {
    const wrapper = shallow(<LinkListMain />)
    expect(toJson(wrapper)).toMatchSnapshot()
  })

  // enzyme doesn't yet support React.createContext
  it.skip('renders a Link in ul for each links in array', () => {
    const props = {
      match: {
        params: {
          category_id: 'THIS_CATEGORY_ID'
        }
      }
    }
    const context = {
      links: [
        {
          "id": "cbc787a0-ffaf-11e8-8eb2-f2801f1b9fd1",
          "name": "Dogs",
          "modified": "2019-01-03T00:00:00.000Z",
          "category_id": "b0715efe-ffaf-11e8-8eb2-f2801f1b9fd1",
          "content": "Corporis accusamus placeat.\n \rUnde."
        },
        {
          "id": "d26e0034-ffaf-11e8-8eb2-f2801f1b9fd1",
          "name": "Cats",
          "modified": "2018-08-15T23:00:00.000Z",
          "category_id": "THIS_CATEGORY_ID",
          "content": "Eos\n \rlaudantium."
        },
        {
          "id": "d26e01a6-ffaf-11e8-8eb2-f2801f1b9fd1",
          "name": "Pigs",
          "modified": "2018-03-01T00:00:00.000Z",
          "category_id": "THIS_CATEGORY_ID",
          "content": "Occaecati dignissimos\nvoluptatum nihil."
        },
        {
          "id": "d26e0570-ffaf-11e8-8eb2-f2801f1b9fd1",
          "name": "Birds",
          "modified": "2019-01-04T00:00:00.000Z",
          "category_id": "b0715efe-ffaf-11e8-8eb2-f2801f1b9fd1",
          "content": "Eum culpa odit."
        },
      ]
    }
    const ul = shallow(<LinkListMain {...props} />, context)
      .find('ul')
    expect(toJson(ul)).toMatchSnapshot()
  })
})
