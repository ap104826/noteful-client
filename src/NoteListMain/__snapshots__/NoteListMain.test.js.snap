// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LinkListMain component renders a .LinkListMain by default 1`] = `
<section
  className="LinkListMain"
>
  <ul />
  <div
    className="LinkListMain__button-container"
  >
    <NavCircleButton
      className="LinkListMain__add-link-button"
      tag={[Function]}
      to="/add-link"
      type="button"
    >
      <FontAwesomeIcon
        border={false}
        className=""
        fixedWidth={false}
        flip={null}
        icon="plus"
        inverse={false}
        listItem={false}
        mask={null}
        pull={null}
        pulse={false}
        rotation={null}
        size={null}
        spin={false}
        symbol={false}
        title=""
        transform={null}
      />
      <br />
      Link
    </NavCircleButton>
  </div>
</section>
`;

exports[`LinkListMain component renders a Link in ul for each links in array 1`] = `
<ul>
  <li
    key="cbc787a0-ffaf-11e8-8eb2-f2801f1b9fd1"
  >
    <Link
      id="cbc787a0-ffaf-11e8-8eb2-f2801f1b9fd1"
      modified="2019-01-03T00:00:00.000Z"
      name="Dogs"
    />
  </li>
  <li
    key="d26e0034-ffaf-11e8-8eb2-f2801f1b9fd1"
  >
    <Link
      id="d26e0034-ffaf-11e8-8eb2-f2801f1b9fd1"
      modified="2018-08-15T23:00:00.000Z"
      name="Cats"
    />
  </li>
  <li
    key="d26e01a6-ffaf-11e8-8eb2-f2801f1b9fd1"
  >
    <Link
      id="d26e01a6-ffaf-11e8-8eb2-f2801f1b9fd1"
      modified="2018-03-01T00:00:00.000Z"
      name="Pigs"
    />
  </li>
  <li
    key="d26e0570-ffaf-11e8-8eb2-f2801f1b9fd1"
  >
    <Link
      id="d26e0570-ffaf-11e8-8eb2-f2801f1b9fd1"
      modified="2019-01-04T00:00:00.000Z"
      name="Birds"
    />
  </li>
</ul>
`;
